firedrake/preconditioners/asm.py

```diff
@@ -280,6 +293,19 @@ def get_patches(self, V):
         return ises
 
 
+def order_points(mesh_dm, points, ordering_type, prefix):
+    if ordering_type == "natural":
+        return points
+    subgraph = [numpy.intersect1d(points, mesh_dm.getAdjacency(p), return_indices=True)[1] for p in points]
+    ia = numpy.cumsum([0] + [len(neigh) for neigh in subgraph]).astype(PETSc.IntType)
+    ja = numpy.concatenate(subgraph).astype(PETSc.IntType)
+    A = PETSc.Mat().createAIJ((len(points), )*2, csr=(ia, ja, numpy.ones(ja.shape, PETSc.RealType)), comm=COMM_SELF)
+    A.setOptionsPrefix(prefix)
```

Comment:
Prefix unnecessary since you never `A.setFromOptions()`. I think you do everything programmatically so there's no need to set the prefix (or pass it in).